akka-stream/src/main/boilerplate/akka/stream/scaladsl/ZipLatestWithApply.scala.template

```diff
@@ -80,11 +83,12 @@ class ZipLatestWith1[[#A1#], O] (val zipper: ([#A1#]) => O) extends GraphStage[F
         ]
       }
 
-      private class ZipLatestInlet[T](in: Inlet[T]) extends InHandler {
+      private class ZipLatestInlet[T](in: Inlet[T], index: Int) extends InHandler {
          var value: T = _
          var hasValue = false
 
          override def onPush() = {
+              if (index == ##0) contextPropagation.suspendContext()
```

Comment:
Maybe add a comment, that only one context can be propagated and we chose the first element as an arbitrary but deterministic choice.